src/MessageTrait.php

```diff
@@ -112,7 +112,7 @@ public function getHeaders()
      */
     public function hasHeader($header)
     {
-        return array_key_exists(strtolower($header), $this->headerNames);
+        return isset($this->headerNames[strtolower($header)]);
```

Comment:
The original was more correct. 

The ABNS for header *values* is:

```text
field-value    = *( field-content / obs-fold )
```

The above means **zero or more** of valid field content/folded content. In other words, you can have a header with no value associated with it.

Since `isset()` will return `false` on an empty string, empty array, or null value, the above would result in a false negative in the case of a header being present with no value.